homeassistant/components/hydrawise/__init__.py

```diff
@@ -50,8 +53,16 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
 async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
     """Set up Hydrawise from a config entry."""
-    access_token = config_entry.data[CONF_API_KEY]
-    hydrawise = legacy.LegacyHydrawiseAsync(access_token)
+
+    if config_entry.data[CONF_USERNAME] == "" or config_entry.data[CONF_PASSWORD] == "":
+        # In order to move to the new GraphQL API, users need to re-authenticate
+        # using the username/password instead of the API key.
+        raise ConfigEntryAuthFailed
+
+    hydrawise = client.Hydrawise(
+        client.Auth(config_entry.data[CONF_USERNAME], config_entry.data[CONF_PASSWORD])
```

Comment:
Don't use `Auth` from the `client` module, it should be imported from `auth`.

Reply:
Fixed.